[BE] Refactor repeated assets in test_foreach.py

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112348
* #112349

Tested conditions in `test_binary_op_list_error_cases` looks almost identical, although it tests method and in-place variants. Use for loop to make distinction a bit more explicit

malfet committed:
[ForEach] Don't test twice, it's all right

Not sure why, but looks like all conditions in `test_binary_op_list_error_cases` is checked twice, I assume due to some merge conflict at some point. Deleted duplicates

[ghstack-poisoned]

janeyx99 reviewed (approved):
what the…how did this happen?

janeyx99 reviewed (changes requested):
OH WAIT I TAKE BACK MY APPROVE. the second set is for the inplace ops, note the _

malfet committed:
Update on "[ForEach] Don't test twice, it's all right"


Not sure why, but looks like all conditions in `test_binary_op_list_error_cases` is checked twice, I assume due to some merge conflict at some point. Deleted duplicates

[ghstack-poisoned]

malfet committed:
Update on "[ForEach] Don't test twice, it's all right"


Not sure why, but looks like all conditions in `test_binary_op_list_error_cases` is checked twice, I assume due to some merge conflict at some point. Deleted duplicates

[ghstack-poisoned]

malfet committed:
Update on "[BE] Refactor repeated assets in test_foreach.py"


Tested conditions in `test_binary_op_list_error_cases` looks almost identical, although it tests method and in-place variants. Use for loop to make distinction a bit more explicit

[ghstack-poisoned]

malfet commented:
@pytorchbot merge
